### DISCUSSION TITLE
Forecasting with SVM

### DISCUSSION THREAD

**Discussion by chemalug** [microsoft/ML-For-Beginners#637] [Q&A]
Hello and thank you very much for the contribution. My question is with reference to forecasting using SVM, since how could you forecast 15 days into the future without knowing the x_test variable?

**Comment by XavierRolex** [comment]
Thank you for your question @chemalug 
In time series forecasting using Support Vector Machines (SVM), we typically train the model on past data to learn patterns and then predict future values based on those patterns.
To forecast 15 days into the future without initially knowing x_test, a recursive (or autoregressive) forecasting approach is often used:
Train the SVM on known historical data (e.g., using a sliding window of past n days to predict the next day).
Predict the next day (t+1) using the latest available input window.
Append this prediction to the input and shift the window forward by one.
Repeat the process step-by-step to generate forecasts for up to 15 days.
So, while x_test may not be fully known ahead of time, it is dynamically built using the model's own prior predictions. This is common in time series models where we don't have ground truth inputs for future timestamps.

**Comment by Hamza-farid** [comment]
That’s a great question. In time series forecasting with SVM (or similar supervised models), you typically don’t have future x_test values available. The common approach is to use past data to generate features that represent patterns (such as lag values, rolling averages, or other engineered time-dependent features). Once the model is trained, you can forecast step by step: predict the next time point using the latest available features, then feed that prediction back as input to generate the next forecast, and so on, until you reach the desired horizon (e.g., 15 days).

This iterative or recursive strategy is one of the standard ways to extend supervised models like SVM to forecasting tasks.
